packages/diagrams/frontend/sirius-components-diagrams/src/representation/DiagramRepresentation.tsx

```diff
@@ -72,9 +72,28 @@ export const DiagramRepresentation = memo(
     const [state, setState] = useState<DiagramRepresentationState>({
       id: crypto.randomUUID(),
       message: null,
+      toolSelections: new Map<string, Selection>(),
     });
     const { addErrorMessage } = useMultiToast();
 
+    const registerPostToolSelection = (id: string, selection: Selection) => {
+      setState((prevState) => {
+        const newToolSelections = new Map(prevState.toolSelections);
+        newToolSelections.set(id, selection);
+        return { ...prevState, toolSelections: newToolSelections };
+      });
+    };
+
+    const consumePostToolSelection = (id: string): Selection | undefined => {
+      const newToolSelections = new Map(state.toolSelections);
+      const selection = newToolSelections.get(id);
+      newToolSelections.delete(id);
+      setState((prevState) => {
+        return { ...prevState, toolSelections: newToolSelections };
+      });
```

Comment:
I suspect that this would trigger a re-render even if we don't have any selection, I'll just add a small check not to perform the setState if we didn't have any selection and thus nothing to delete nor to update because update the state of such a high level component may re-render a lot of things.